### ISSUE TITLE
Parallel model loading on multiple GPUs on startup

### ISSUE THREAD

**Issue by leshanbog** [triton-inference-server/server#4367]
I have one model per GPU (no tensor or pipeline parallelism) and multiple GPUs. When starting tritonserver, the models are loaded sequentially: first model on the first GPU, only after it is done with first, tritonserver begins to allocate on the second GPU, ... 

_Is there a way to load them in parallel?_
It would substantially speed up the server startup time.

**Comment by nv-kmcgill53** [comment]
Hi @leshanbog, 
> I have one model per GPU (no tensor or pipeline parallelism) and multiple GPUs.
For clarity, does this mean you have multiple GPUs available on the machine and there is one model on each GPU? If so, can you please re-order the model's device settings to start on different GPUs to confirm your observation? Thank you.

**Comment by leshanbog** [comment]
Hi @nv-kmcgill53,

> For clarity, does this mean you have multiple GPUs available on the machine and there is one model on each GPU?

Yes, exactly.

Reordering affects the order with which models are loaded on each GPU, but it is still done sequentially. I make reordering by changing the order in instance_group option in config.pbtxt. E.g.

```
instance_group [
  {
    count: 1
    kind: KIND_GPU
    gpus: [ 1 ]
  },
  {
    count: 1
    kind: KIND_GPU
    gpus: [ 0 ]
  }
]
```

By the way, at the beginning tritonserver allocates ~ 400 MB on each GPU in parallel, but the rest ~ 13 GB are loaded sequentially.

**Comment by CoderHam** [comment]
This is a common ask I have seen from multiple users. @nv-kmcgill53 can you file a ticket for this so the team an evaluate the same? marking as an enhancement for now

**Comment by nv-kmcgill53** [comment]
Ticket created

**Comment by nv-kmcgill53** [comment]
#4303

**Comment by Tabrizian** [comment]
This feature has been implemented by @kthui. Closing the bug. Feel free to open additional issues if you have any comments/questions. It is available in Triton 23.04+.

**Comment by priyankat99** [comment]
can't find any information how to use this feature? would be really helpful in speeding up start up times 
 the linked issue links back to this issue
